sw/device/tests/sram_ctrl_execution_test_main.c

```diff
@@ -0,0 +1,163 @@
+ * IFETCH_OFFSET = (x - y) / 8 = 8
+ */
+static volatile const uint32_t kOtpIfetchHwRelativeOffset = 8;
+
+/*
+ * Buffer to hold instructions to be executed.
+ *
+ * Because this is the Main SRAM, we let the compiler allocate the buffer
+ * instead of using the hard-coded locations from the `sram_ctrl_testutil.c`,
+ * in order to prevent clobbering the data used by the program.
+ *
+ * This can be thought of as the implementation for the `ram_exec_function_t`.
+ *
+ * NOTE: We explicitly specify the `.data` section, otherwise due to the
+ *       `const` qualifier buffer would end up in flash. The implicit
+ *       alternative would be to drop the `const` qualifier, making the buffer
+ *       go into the `.data` section.
+ */
+__attribute__((section(
+    ".data"))) static volatile const uint32_t execution_test_instructions[2] = {
+    /*
+     *   Abstract representation: add rd, rs1, rs2
+     *   +-----------+-------+-------+-------+------+-----------+
+     *   |  0000000  |  rs2  |  rs1  |  000  |  rd  |  0110011  |
+     *   +-----------+-------+-------+-------+------+-----------+
+     *   31        25 24   20 19   15 14   12 11   7 6          0
+     *
+     *   Custom representation: add a5, a0, a5
+     *   +-----------+---------+---------+-------+--------+-----------+
+     *   |  0000000  |  01010  |  01011  |  000  | 01010  |  0110011  |
+     *   +-----------+---------+---------+-------+--------+-----------+
+     *   31        25 24     20 19     15 14   12 11     7 6          0
+     */
+    0x00A58533,
+    /*
+     *   Abstract representation: ret == jalr rd, offset(rs1)
+     *   +----------------+-------+-------+------+-----------+
+     *   |  000000000000  |  rs1  |  000  |  rd  |  1100111  |
+     *   +----------------+-------+-------+------+-----------+
+     *   31             20 19   15 14   12 11   7 6          0
+     *
+     *   Custom representation: jalr zero, 0(ra)
+     *   +----------------+---------+-------+---------+-----------+
+     *   |  000000000000  |  00001  |  000  |  00000  |  1100111  |
+     *   +----------------+---------+-------+---------+-----------+
+     *   31             20 19     15 14   12 11      7 6          0
+     */
+    0x00008067,
+};
+
+/*
+ * Adds the first parameter to the second parameter and returns the result.
+ *
+ * Maps to the `execution_test_instructions` buffer as the function body.
+ */
+typedef uint32_t (*ram_exec_function_t)(volatile uint32_t, volatile uint32_t);
+
+static bool otp_ifetch_enabled(void) {
+  otp_ctrl_t otp = otp_ctrl_init();
+
+  dif_otp_ctrl_config_t config = {
+      .check_timeout = 100000,
+      .integrity_period_mask = 0x3ffff,
+      .consistency_period_mask = 0x3ffffff,
+  };
+  CHECK_DIF_OK(dif_otp_ctrl_configure(&otp.dif, config));
+
+  CHECK_DIF_OK(dif_otp_ctrl_dai_read_start(&otp.dif, kDifOtpCtrlPartitionHwCfg,
+                                           kOtpIfetchHwRelativeOffset));
+
+  otp_wait_for_dai(&otp);
```

Comment:
can replace with spin wait macro directly. See my comment [here](https://github.com/lowRISC/opentitan/pull/8941#discussion_r739494124).

Reply:
Haven't changed this yet, pending response on the original PR.

Reply:
Keeping, as per the original pr.